src/Microsoft.DotNet.Darc/DarcLib/Helpers/DependencyFileManager.cs

```diff
@@ -211,6 +211,84 @@ public async Task AddDependencyAsync(
         await AddDependencyToVersionDetailsAsync(repoUri, branch, dependency);
     }
 
+    public async Task RemoveDependencyAsync(DependencyDetail dependency, string repoUri, string branch, bool repoIsVmr = false)
+    {
+        var updatedDependencyVersionFile =
+            new GitFile(VersionFiles.VersionDetailsXml, await RemoveDependencyFromVersionDetailsAsync(dependency, repoUri, branch));
+        var updatedVersionPropsFile =
+            new GitFile(VersionFiles.VersionProps, await RemoveDependencyFromVersionPropsAsync(dependency, repoUri, branch));
+        List<GitFile> gitFiles = [updatedDependencyVersionFile, updatedVersionPropsFile];
+
+        var updatedDotnetTools = await RemoveDotnetToolsDependencyAsync(dependency, repoUri, branch, repoIsVmr);
+        if (updatedDotnetTools != null)
+        {
+            gitFiles.Add(new(VersionFiles.DotnetToolsConfigJson, updatedDotnetTools));  
+        }
+
+        await GetGitClient(repoUri).CommitFilesAsync(
+            gitFiles,
+            repoUri,
+            branch,
+            $"Remove {dependency.Name} from Version.Details.xml and Version.props'");
+
+        _logger.LogInformation($"Dependency '{dependency.Name}' successfully removed from '{VersionFiles.VersionDetailsXml}'");
+    }
+
+    private async Task<JObject> RemoveDotnetToolsDependencyAsync(DependencyDetail dependency, string repoUri, string branch, bool repoIsVmr)
+    {
+        var dotnetTools = await ReadDotNetToolsConfigJsonAsync(repoUri, branch, repoIsVmr);
+        if (dotnetTools != null)
+        {
+            var tools = dotnetTools["tools"] as JObject;
+            if (tools != null)
+            {
+                // we have to do this because JObject is case sensitive
+                var toolProperty = tools.Properties().FirstOrDefault(p => p.Name.Equals(dependency.Name, StringComparison.OrdinalIgnoreCase));
+                if (toolProperty != null)
+                {
+                    tools.Remove(toolProperty.Name);
+                }
+
+                return dotnetTools;
+            }
+        }
+
+        return null;
```

Comment:
Can we swap the `if` and reduce nesting?

Reply:
done, VS also did some fancy pattern matching stuff